sensorium/training/trainers.py

```diff
@@ -109,7 +160,17 @@ def full_objective(model, dataloader, data_key, *args, **kwargs):
 
     n_iterations = len(LongCycler(dataloaders["train"]))
 
-    optimizer = torch.optim.Adam(model.parameters(), lr=lr_init)
+    if include_kldivergence:
+        cluster_centers = torch.zeros(
+            cluster_number, 128, dtype=torch.float, requires_grad=True
+        )
```

Comment:
```suggestion
        k = list(model.readout.keys())[0]
        dim = model.readout[k].features.shape[1]
        dtype = model.readout[k].features.dtype
        cluster_centers = torch.zeros(
            cluster_number, dim, dtype=dtype, requires_grad=True
        )
```
please make sure that this works!